File: Bua.CodeRev.CompilerService/Bua.CodeRev.CompilerService.Core/Models/ExecutionResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Bua.CodeRev.CompilerService.Core.Models
{
    public class ExecutionResult
    {
        public bool Success { get; set; }
        public IEnumerable<string> Output { get; set; }
        public IEnumerable<CompilationError> Errors { get; set; }
    }
}

File: Bua.CodeRev.WebClient/src/app/code-editor/models/executionResult.ts
import { CompilationError } from './compilationError';
import { ExecutionResultResponse } from './response/executionResult-response';

export class ExecutionResult {
    public success: boolean;
    public output: string[];
    public errors: CompilationError[];

    constructor(resp: ExecutionResultResponse) {
        this.success = resp.success ?? false;
        this.output = resp.output ?? ['Не был получен ответ с сервера'];
        this.errors = resp.errors ?? [];
    }
}